server/osx/server_osx.py

```diff
@@ -456,14 +453,19 @@ def key_press(
 
 def write_text(text, paste=False):
     '''send text formatted exactly as written to active window.  will use
-       pbpaste clipboard to paste the text instead of typing it.'''
+       simulate keypress typing for maximum compatibility.'''
 
-    logging.debug("text = %s paste = %s" % (text, paste))
+    logging.debug("text = %s" % (text))
     if text:
-        # copy the pasted text to the clipboard
-        write_command(text, arguments='', executable='pbcopy')
-        # paste
-        key_press('v', 'super')
+        script = applescript.AppleScript('''
+        tell application "System Events"
+          repeat with i from 1 to count characters of "{text}"
+            keystroke (character i of "{text}")
+            delay 0.0002
+          end repeat
+        end tell
+        '''.format(text=text))
+        script.run()
```

Comment:
@seananderson ❤️

Reply:
Note that you only need `keystroke "{text}"` to use simulated keystrokes instead of pasting; however, about a year ago sometimes keystrokes started getting missed. As far as I can tell, some update to the operating system changed this. That's when I added a micro delay and iterated over each character. https://github.com/seananderson/aenea/commit/5e8a6dec8b267f70e2392d19f502659f8119559e

It's possible this is unique to my set up or that other people would need a different delay. Obviously, it's a bit faster if you can avoid the delay.

Reply:
Daaang `keystroke "{text}"` Is so much faster!

I have not been able to replicate your missing keystrokes bug, maybe Apple fixed as it has been a year. (I am running 10.13.6 by the way). 

Would you be able to see if running the script below capitalises words correctly? Just beware that this will start typing stuff after two seconds so make sure you have a text field open and selected after clicking run on the script editor

```applescript
delay 2
tell application "System Events"
	keystroke "# 1: Testing hello. This is a test to Test that the words are capitalised Correctly"
	keystroke "# 2: Testing hello. This is a test to Test that the words are capitalised Correctly"
	keystroke "# 3: Testing hello. This is a test to Test that the words are capitalised Correctly"
end tell
```

Reply:
`keystroke "{text}"` seems to be working for me now. I'd forgotten just how much faster this is!

Your test does print the words with the proper capitalization for me. If doesn't for you then it might have to do with the auto correction and auto capitalization settings in the operating system ("Correct spelling automatically" and "Capitalize words automatically" in the Keyboard Preferences pane). I think it was at the time that those were introduced that I first had the problems. I have them turned off.

Reply:
I have tried turning on the capitalise words automatically and correct spelling automatically options and can't replicate the bug. I think i will put it back to `keystroke "{text}"` and submit another pull request if I happen to replicate the but somehow